hashintel/hash
# Normalize URLs and better error handling when loading remote blocks

## 🌟 What is the purpose of this PR?

Introduce a few improvements to the experience when using the input to load a block from a URL the user inputs.

## 🔍 What does this change?
- Overwrite the _local storage_ cache for a block when the `source` key changes (in case a block is rebuilt locally with a new source file, without the version number being bumped)
- Normalize the URL used to load a block, to avoid the same block under different keys
- Deletes the block-fetching-promise from the _in memory_ cache for block promises (`blockCache`) if the promise fails, to allow a user to correct an error without having to reload the app (previously a failed promise would remain in the cache, preventing retry).

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201959586244678/1202057575279137/f) _(internal)_

## ❓ How to test this?



1.  Have a block served from some URL, which isn't already in the app 
2. Check URL normalization by introducing trailing slash or whitespace
3. Check cache-busting by first trying the URL when it is not being served, and then again when it is - it should load the second time.
